### PR TITLE
New package: leftwm-0.2.11

### PR DESCRIPTION
**Pull request by uncomfyhalomacro**
#### General
- [x] This is a new package and it conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements)

#### Have the results of the proposed changes been tested?
- [x] I use the packages affected by the proposed changes on a regular basis and confirm this PR works for me
- [ ] I generally don't use the affected packages but briefly tested this PR

#### Does it build and run successfully? 
(Please choose at least one native build and, if supported, at least one cross build. More are better.)
- [x] I built this PR locally for my native architecture, (ARCH-LIBC)
- [x] I built this PR locally for these architectures (if supported. mark crossbuilds):
  - [x] aarch64
  - [x] aarch64-musl
  - [x] x86_64-musl
  - [x] i686
  - [ ] armv7l
  - [ ] armv6l
  - [ ] armv6l-musl

#### Questions:
- ~~Should I include `x86_64-musl`? It builds correctly but it does not run correctly. See https://github.com/leftwm/leftwm/issues/396 . I did not include it since I do not want the users to experience a broken install.~~

### CODE REVIEW

File: srcpkgs/leftwm/template
Comment (Johnnynator) [inline srcpkgs/leftwm/template:5]: Do not limit archs.
If it is broken on musl add
```
if [ "$XBPS_TARGET_LIBC" = musl ] ; then
	broken="Some reason/upstream issue link"
fi
```

File: srcpkgs/leftwm/INSTALL.msg
Comment (Johnnynator) [inline srcpkgs/leftwm/INSTALL.msg:1]: There is no reason, why the package manager should inform you about this type of documentation.
This is a tiling WM, you are usually not using it without reading the documentation at all, and you provide the `README.md` in the package which already has the information about `config.toml`

File: srcpkgs/leftwm/template
Comment (ReneganRonin) [reply srcpkgs/leftwm/template]: alright. i am very new to packaging in void. I did not know that exists

Comment (ericonr) [reply srcpkgs/leftwm/template]: It isn't clear to me from your issue, did you use the package built from within xbps-src or did you build the software manually and try to use it? If the latter, you're probably not using the Void toolchain with the necessary patches.

Furthermore, please include a link to the issue in the broken definition.

Comment (ReneganRonin) [reply srcpkgs/leftwm/template]: i did not try Void's toolchain. I once dual boot with x86_64-musl of Void and tried it for a bit (left it because there are some stuff I need to work properly), installed rustup and cargo from the website and build leftwm. it did not work. i assumed they were the same so I mentioned that it has trouble with musl, i was wrong. thanks for letting me know that Void has patches for it.

I guess I will remove that line and that if block then. and let the CI do its thing.

**Comment by Johnnynator** [comment]
Surpise, a rust thingie that is broken on 32bit systems.

**Comment by ReneganRonin** [comment]
so i am only going to list the ones on `archs=` from those that are built successfully right?

**Comment by Johnnynator** [comment]
> so i am only going to list the ones on `archs=` from those that are built successfully right?

```
if [ "$XBPS_TARGET_WORDSIZE" = 32 ]; then
	broken="Another rust project that doesn't get c <-> rust types compatibility correct"
fi
```

and maybe ask upstream nicely to fix it.

**Comment by ReneganRonin** [comment]
lol i forgot to delete that `archs=` line

**Comment by ericonr** [comment]
I feel like you're missing some makedepends, this should link against some X11 library, I *think*.

**Comment by ReneganRonin** [comment]
yeah i was thinking of that last time. but i think not since build dependencies is not provided in the README. i might be wrong, i should ask the upstream maintainer about that.

**Comment by ReneganRonin** [comment]
it seems from what i can see in their CI, they only have `libsystemd-dev` (which i think is not available in Void since it has a different init) but I am sure it is optional and use for some features of the package. i will ask just to be sure.

**Comment by ReneganRonin** [comment]
if we look at https://github.com/leftwm/leftwm#dependencies, i dont think there are any. also they have not replied yet. I am going to wait.

**Comment by ReneganRonin** [comment]
Update: Upstream maintainer told me that cargo will handle all deps so there is no need to put something in `makedepends` 😄

**Comment by ReneganRonin** [comment]
> > so i am only going to list the ones on `archs=` from those that are built successfully right?
> 
> ```
> if [ "$XBPS_TARGET_WORDSIZE" = 32 ]; then
> 	broken="Another rust project that doesn't get c <-> rust types compatibility correct"
> fi
> ```
> 
> and maybe ask upstream nicely to fix it.

i forgot to mention that they have a fix upstream but i think that will be in the next future release

**Comment by ericonr** [comment]
If feasible, importing as a patch is an option (or asking for a patch release).

**Comment by ReneganRonin** [comment]
Update: It seems 0.2.8 release is closer than I thought (probably this week or next week) so I will be waiting for that with the added patch :D

**Comment by ReneganRonin** [comment]
I am testing now the new release. it seems it builds correctly with 32 bit archs. now testing armv7 and then armv* musl

**Comment by ReneganRonin** [comment]
done

**Comment by VuiMuich** [comment]
Hi,
as we just today had a question regarding leftwm on void, I went through our issue tracker and followed to this PR.
If this PR is still actively pursued, here is a heads up that we are about to release version `0.2.10` of `leftwm` in the next few weeks.
Thanks for your efforts to make leftwm available for a wider user base.

File: srcpkgs/leftwm/template
Comment (ericonr) [inline srcpkgs/leftwm/template:15]: Remember to clean this up, as well.

**Comment by 0323pin** [comment]
> I feel like you're missing some makedepends, this should link against some X11 library, I _think_.

pkgsrc maintainer and Void musl user here,
You're right, see this: https://github.com/leftwm/leftwm/discussions/590

**Comment by uncomfyhalomacro** [comment]
> > I feel like you're missing some makedepends, this should link against some X11 library, I _think_.
> 
> 
> 
> pkgsrc maintainer and Void musl user here,
> 
> You're right, see this: https://github.com/leftwm/leftwm/discussions/590

Oh thanks!! I will take a look on the said issue and add the needed x11 libraries/dependences for the template. I will do it once 0.2.10 lands 😄

**Comment by 0323pin** [comment]
@uncomfyhalomacro I'm running `leftwm` built from git-HEAD on NetBSD right now. You will need Rust >= 1.56 to build 0.2.10, Void is still shipping Rust-1.53

**Comment by VuiMuich** [comment]
> @uncomfyhalomacro I'm running `leftwm` built from git-HEAD on NetBSD right now. You will need Rust >= 1.56 to build 0.2.10, Void is still on Rust-1.53

As of now we haven't pushed the 2021 Edition Upgrade.
Do you see realistic chances of Void bumping their Rust version in close future? We might consider holding this back as we currently are not depending on any of the new optimizations or features.

**Comment by uncomfyhalomacro** [comment]
> > @uncomfyhalomacro I'm running `leftwm` built from git-HEAD on NetBSD right now. You will need Rust >= 1.56 to build 0.2.10, Void is still on Rust-1.53
> 
> As of now we haven't pushed the 2021 Edition Upgrade for `leftwm`. Do you see realistic chances of Void bumping their Rust version in close future? We might consider holding this back as we currently are not depending on any of the new optimizations or features.
> 
> Tagging [leftwm/leftwm#594](https://github.com/leftwm/leftwm/pull/594)

Hello. It has been a while. I already updated the template to comply with the dependencies mentioned on the README. Hoping this gets merged. :)

File: srcpkgs/leftwm/template
Comment (Johnnynator) [inline srcpkgs/leftwm/template:24]: Just do everything in do_install.

Otherwise template looks good

**Review by Johnnynator — CHANGES_REQUESTED** [review]

**Comment by uncomfyhalomacro** [comment]
hmm weird. did i make a wrong environmental variable? is using XBPS_RUST_TARGET wrong?

**Comment by uncomfyhalomacro** [comment]
@Johnnynator all checks have passed. Can you please review if there are stuff that I might have missed? :D Thanks again!

**Comment by Johnnynator** [comment]
I simplified your vbin stuff to just always `{RUST_TARGET}` and removed the for loop.

**Comment by 0323pin** [comment]
@uncomfyhalomacro `leftwm`-0.3.0 has been released, I've merged the update on NetBSD some hours ago. Would you mind updating here? I use it on both my machines and it's nice when the config files are compatible.

TIA

**Comment by uncomfyhalomacro** [comment]
i will try to do it tomorrow i guess if i have time. im currently busy for a few days in uni so yeah

**Pull request merged**
